### PR TITLE
Fix various Sonar warnings

### PR DESCRIPTION
**Pull request by 304NotModified**
Used R# for this

This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/3312)

### CODE REVIEW

File: src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ protected override void RenderInnerAndTransform(LogEventInfo logEvent, StringBui
                 {
                     string str = builder.ToString(orgLength, builder.Length - orgLength);
                     str = str.Replace(newLine, Replacement);
-                    if (newLine != "\n" && !(Replacement?.IndexOf('\n') >= 0) && str.IndexOf('\n') >= 0)
+                    if (newLine != "\n" && Replacement?.IndexOf('\n') < 0 && str.IndexOf('\n') >= 0)
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs]: @snakefoot could you please double check this change? Thanks!

File: src/NLog/MappedDiagnosticsContext.cs [hunk 2]
```diff
@@ -52,10 +52,7 @@ public static class MappedDiagnosticsContext
 
         private static readonly IDictionary<string, object> EmptyDefaultDictionary = new SortHelpers.ReadOnlySingleBucketDictionary<string, object>();
 
-        /// <summary>
-        /// 
-        /// </summary>
-        private class ItemRemover : IDisposable
+        private sealed class ItemRemover : IDisposable
```
Comment (304NotModified) [inline src/NLog/MappedDiagnosticsContext.cs]: note: sealed for keep Sonar stop warning about IDisposable

File: src/NLog/Targets/TargetWithContext.cs [hunk 3]
```diff
@@ -516,7 +509,7 @@ protected virtual bool SerializeMdcItem(LogEventInfo logEvent, string name, obje
             foreach (var name in names)
             {
                 object value = MappedDiagnosticsLogicalContext.GetObject(name);
-                if (SerializeMdlcItem(logEvent, name, value, out var serializedValue))
+                if (SerializeMdlcItem(logEvent, name, value, out _))
```
Comment (304NotModified) [inline src/NLog/Targets/TargetWithContext.cs]: @snakefoot is this correct we aren't using the serializedValue? (2 times)

Comment (snakefoot) [reply src/NLog/Targets/TargetWithContext.cs]: No that is a bug.

File: src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs]: Hehe I suck at null-conditional operators :)

Comment (304NotModified) [reply src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs]: yeah, doubting if its the same, maybe have to use Replacement!=null explicit,

File: src/NLog/Layouts/JsonLayout.cs [hunk 4]
```diff
@@ -80,9 +80,9 @@ public bool SerializeObject(object value, StringBuilder builder)
                     return _converter.SerializeObject(value, builder);
             }
 
-            public bool SerializeObjectNoLimit(object value, StringBuilder builder)
+            public void SerializeObjectNoLimit(object value, StringBuilder builder)
```
Comment (snakefoot) [inline src/NLog/Layouts/JsonLayout.cs]: You are touching public methods. Breaking change.

Comment (snakefoot) [reply src/NLog/Layouts/JsonLayout.cs]: Nope just ignore me

File: src/NLog/Targets/Wrappers/FallbackGroupTarget.cs [hunk 5]
```diff
@@ -127,11 +130,13 @@ protected override void Write(AsyncLogEventInfo logEvent)
                         // success
                         if (ReturnToFirstOnSuccess)
                         {
+#pragma warning disable S1066 // Collapsible "if" statements should be merged
 #if !SILVERLIGHT || WINDOWS_PHONEs
```
Comment (snakefoot) [inline src/NLog/Targets/Wrappers/FallbackGroupTarget.cs]: Funny with `WINDOWS_PHONEs`

File: src/NLog/Targets/TargetWithContext.cs (hunk 3 above)
Comment (304NotModified) [reply src/NLog/Targets/TargetWithContext.cs]: OK fixing this now

File: src/NLog/Layouts/JsonLayout.cs (hunk 4 above)
Comment (304NotModified) [reply src/NLog/Layouts/JsonLayout.cs]: 👼 That's the reason I don't like the keyword `public` in internal types ;)

File: src/NLog/Targets/Wrappers/FallbackGroupTarget.cs (hunk 5 above)
Comment (304NotModified) [reply src/NLog/Targets/Wrappers/FallbackGroupTarget.cs]: yes good know, also nice it just compiles.

Comment (304NotModified) [reply src/NLog/Targets/Wrappers/FallbackGroupTarget.cs]: not sure what to do about this (anyway, not in the PR)

What do you think?

Comment (snakefoot) [reply src/NLog/Targets/Wrappers/FallbackGroupTarget.cs]: Created #3320

**Pull request merged**